File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
    HashMap<Character,Integer> h=new HashMap<>();
    for(int i=0;i<s.length();i++)
    {
        h.put(s.charAt(i),h.getOrDefault(s.charAt(i),0)+1);
    }
    for(int i=0;i<t.length();i++)
    {
        if(h.containsKey(t.charAt(i)))
        { 
         int x=h.get(t.charAt(i))-1;
         if(x==0)
         {
          h.remove(t.charAt(i));
         }
         else if(x>0)
          h.put(t.charAt(i),x);
         else 
          return false;
        }
        else
         return false;  
    }
 return h.isEmpty();   
 }      
}

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
         return False

        dic = {}

        for c in s:
            dic[c] = dic.get(c,0)+1

        for c in t:
            if c not in dic:
                return False
            dic[c]=dic[c]-1
            if dic[c]==0:
                del dic[c]

        return len(dic) == 0     


        